modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2770,37 +2748,33 @@ private CompletableFuture<CompletableFuture<?>> applyUpdateCommand(
 
                 applyCmdWithExceptionHandling(cmd, resultFuture);
 
-                return resultFuture.thenApply(res -> {
+                return resultFuture.thenCompose(res -> {
                     UpdateCommandResult updateCommandResult = (UpdateCommandResult) res;
 
                     if (full && updateCommandResult != null && !updateCommandResult.isPrimaryReplicaMatch()) {
```

Comment:
Do we still need `updateCommandResult != null` check?
And you removed `full` condition from `applyUpdateAllCommand` and left it here

Reply:
> Do we still need updateCommandResult != null check?

Probably not. Removed.

> Also, full condition is not needed

Why?

Reply:
Agree. Removed.